app/src/main/java/com/example/drawingactivity/DialogUtils.kt

```diff
@@ -137,6 +138,141 @@ object DialogUtils {
+        val gradientPicker = view.findViewById<GradientView>(R.id.gradientPicker)
+        val hexText = view.findViewById<ForcableEditText>(R.id.hexText)
+        val hueText = view.findViewById<ForcableEditText>(R.id.hueText)
+        val satText = view.findViewById<ForcableEditText>(R.id.satText)
+        val valText = view.findViewById<ForcableEditText>(R.id.valText)
+        val redText = view.findViewById<ForcableEditText>(R.id.redText)
+        val greenText = view.findViewById<ForcableEditText>(R.id.greenText)
+        val blueText = view.findViewById<ForcableEditText>(R.id.blueText)
+        val topColorSquare = view.findViewById<View>(R.id.topColor)
+        val bottomColorSquare = view.findViewById<View>(R.id.bottomColor)
+        val confirmButton = view.findViewById<Button>(R.id.okButton)
+        val allText = listOf(hexText, hueText, satText, valText, redText, greenText, blueText)
+        confirmButton.setOnClickListener {
+            onSubmit.invoke(
+                Hsv(
+                    hueText.floatValue,
+                    satText.floatValue / 100,
+                    valText.floatValue / 100
+                )
+            )
+            sheet.dismiss()
+        }
+        val cancelButton = view.findViewById<Button>(R.id.cancelButton)
+        cancelButton.setOnClickListener {
+            onSubmit.invoke(null)
+            sheet.dismiss()
+        }
+        sheet.setOnCancelListener {
+            onSubmit.invoke(null)
+            sheet.dismiss()
+        }
+        /**
+         * Lambda that updates all the views in the gradient.
+         * Takes in a color state data class to update views.
+         */
+        val forceViewUpdates: ((Hsv) -> Unit) = { hsv ->
+            val rgb = calculateColorWithHsv(hsv)
+            val hex = calculateHex(rgb)
+            val saturation = hsv.saturation * 100
+            val value = hsv.value * 100
+            hexText.forceText(hex)
+            hueText.forceText(hsv.hue.toInt().toString())
+            satText.forceText(saturation.toInt().toString())
+            valText.forceText(value.toInt().toString())
+            redText.forceText(rgb[0].toString())
+            greenText.forceText(rgb[1].toString())
+            blueText.forceText(rgb[2].toString())
+            huePicker.setHue(hsv.hue)
+            gradientPicker.setHue(hsv.hue)
+            gradientPicker.setSaturationAndValue(hsv.saturation, hsv.value)
+            topColorSquare.setBackgroundColor(parseColor(hex))
+        }
+        if (previousColor != null) {
+            val rgb = calculateColorWithHsv(previousColor)
+            val hex = calculateHex(rgb)
+            bottomColorSquare.setBackgroundColor(parseColor(hex))
+            forceViewUpdates.invoke(previousColor)
+        }
+        listOf(redText, greenText, blueText).forEach {
+            it.addTextChangedListener {
+                if (redText.checkIfInt != null && greenText.checkIfInt != null && blueText.checkIfInt != null) {
```

Comment:
`checkIfInt` is an odd name to return a nullable. You'd think by the name it'd return true or false. Clean up naming